### PR TITLE
WWT-4876 - add method searchAccounts() to multichain

### PR DESCRIPTION
**Pull request by denshuliar**

### CODE REVIEW

File: src/BladeSDK.ts [hunk 1]
```diff
@@ -708,23 +709,24 @@ export class BladeSDK {
      * @param keyOrMnemonic BIP39 mnemonic, private key with DER header
      * @param completionKey optional field bridge between mobile webViews and native apps
      * @returns {AccountPrivateData}
+    */
+
     async searchAccounts(keyOrMnemonic: string, completionKey?: string): Promise<AccountPrivateData> {
         try {
             const accounts: AccountPrivateRecord[] = [];
             if (keyOrMnemonic.trim().split(" ").length > 1) {
```
Comment (Dubiy) [inline src/BladeSDK.ts]: move this logic to AccountServiceHedera.ts and leave a single call `this.accountServiceContext.searchAccounts(keyOrMnemonic)`

File: src/BladeSDK.ts [hunk 2]
```diff
@@ -708,23 +709,24 @@ export class BladeSDK {
      * @param keyOrMnemonic BIP39 mnemonic, private key with DER header
      * @param completionKey optional field bridge between mobile webViews and native apps
      * @returns {AccountPrivateData}
+    */
```
Comment (Dubiy) [inline src/BladeSDK.ts:711]: remove `// TODO ...` and empty line

File: src/strategies/AccountServiceContext.ts [hunk 3]
```diff
@@ -36,6 +38,14 @@ export interface IAccountService {
         nextPage: string,
         transactionsLimit: string
     ): Promise<TransactionsHistoryData>;
+    getAccountsFromMnemonic(
```
Comment (Dubiy) [inline src/strategies/AccountServiceContext.ts]: move both methods to AccountServiceHedera. add `searchAccounts`

File: src/BladeSDK.ts [hunk 4]
```diff
@@ -708,23 +709,24 @@ export class BladeSDK {
...
+
     async searchAccounts(keyOrMnemonic: string, completionKey?: string): Promise<AccountPrivateData> {
         try {
             const accounts: AccountPrivateRecord[] = [];
             if (keyOrMnemonic.trim().split(" ").length > 1) {
                 // mnemonic
-                accounts.push(...(await getAccountsFromMnemonic(keyOrMnemonic, this.network)));
+                accounts.push(...(await this.accountServiceContext.getAccountsFromMnemonic(keyOrMnemonic, this.network)));
```
Comment (Dubiy) [inline src/BladeSDK.ts]: no need to pass this.network. There is chainId inside service

File: src/strategies/hedera/AccountServiceHedera.ts [hunk 5]
```diff
@@ -279,6 +282,106 @@ export default class AccountServiceHedera implements IAccountService {
...
+                        key = await mnemonic.toEcdsaPrivateKey();
+                    }
+                } else {
+                    if (standard) {
+                        key = await mnemonic.toStandardEd25519PrivateKey();
+                    } else {
+                        key = await mnemonic.toEd25519PrivateKey();
+                    }
+                }
+                records.push(...(await this.prepareAccountRecord(key, keyType, network)));
```
Comment (Dubiy) [inline src/strategies/hedera/AccountServiceHedera.ts]: may be improved with `Promise.all`, if you want to

**Review by Dubiy — CHANGES_REQUESTED** [review]

File: src/BladeSDK.ts (hunk 2 above)
Comment (denshuliar) [reply src/BladeSDK.ts:711]: fixed ✅

File: src/BladeSDK.ts (hunk 4 above)
Comment (denshuliar) [reply src/BladeSDK.ts]: fixed ✅

File: src/strategies/AccountServiceContext.ts (hunk 3 above)
Comment (denshuliar) [reply src/strategies/AccountServiceContext.ts]: fixed ✅

File: src/BladeSDK.ts (hunk 1 above)
Comment (denshuliar) [reply src/BladeSDK.ts]: fixed ✅

File: src/strategies/hedera/AccountServiceHedera.ts (hunk 5 above)
Comment (denshuliar) [reply src/strategies/hedera/AccountServiceHedera.ts]: fixed ✅

File: src/strategies/hedera/AccountServiceHedera.ts [hunk 6]
```diff
@@ -279,6 +282,126 @@ export default class AccountServiceHedera implements IAccountService {
...
+                    keyPromise = standard 
+                        ? mnemonic.toStandardECDSAsecp256k1PrivateKey() 
+                        : mnemonic.toEcdsaPrivateKey();
+                } else {
+                    keyPromise = standard 
+                        ? mnemonic.toStandardEd25519PrivateKey() 
+                        : mnemonic.toEd25519PrivateKey();
+                }
+    
+                promises.push(keyPromise.then(key => this.prepareAccountRecord(key, keyType)));
```
Comment (Dubiy) [inline src/strategies/hedera/AccountServiceHedera.ts]: let simplify for readability 

```
if (keyType === CryptoKeyType.ECDSA_SECP256K1) {
    key = standard 
        ? await mnemonic.toStandardECDSAsecp256k1PrivateKey() 
        : await mnemonic.toEcdsaPrivateKey();
} else {
    key = standard 
        ? await mnemonic.toStandardEd25519PrivateKey() 
        : await mnemonic.toEd25519PrivateKey();
}
promises.push(this.prepareAccountRecord(key, keyType));
```

File: src/BladeSDK.ts [hunk 7]
```diff
@@ -700,31 +701,23 @@ export class BladeSDK {
...
-    // TODO implement `searchAccounts` method
     /**
      * Get accounts list and keys from private key or mnemonic
      * Returned keys with DER header.
      * EvmAddress computed from ECDSA Public key.
      * @param keyOrMnemonic BIP39 mnemonic, private key with DER header
      * @param completionKey optional field bridge between mobile webViews and native apps
      * @returns {AccountPrivateData}
+    */
+
```
Comment (Dubiy) [inline src/BladeSDK.ts]: remove empty line

**Review by Dubiy — CHANGES_REQUESTED** [review]

File: src/strategies/hedera/AccountServiceHedera.ts (hunk 6 above)
Comment (denshuliar) [reply src/strategies/hedera/AccountServiceHedera.ts]: fixed ✅

**Review by Dubiy — APPROVED** [review]

**Pull request merged**